rust/core-lib/src/editor.rs

```diff
@@ -685,6 +686,48 @@ impl Editor {
         }
     }
 
+    // capitalization behaviour is similar to behaviour in XCode
+    fn capitalize_text(&mut self, view: &mut View) {
+        let mut builder = delta::Builder::new(self.text.len());
+        let mut final_selection = Selection::new();
+
+        for &region in view.sel_regions() {
+            final_selection.add_region(SelRegion::new(region.max(), region.max()));
+            let mut word_cursor = WordCursor::new(&self.text, region.min());
+
+            loop {
+                // capitalize each word in the current selection
+                let (start, end) = word_cursor.select_word();
+
+                if start < end {
+                    let interval = Interval::new_closed_open(start, end);
+                    let word = self.text.slice_to_string(start..end);
+
+                    // first letter is uppercase, remaining letters are lowercase
+                    let (first_char, rest) = word.split_at(1);
+                    let mut capitalized_text = String::from("");
+                    capitalized_text.push_str(&first_char.to_uppercase());
+                    capitalized_text.push_str(&rest.to_lowercase());
+                    builder.replace(interval, Rope::from(capitalized_text));
```

Comment:
I wonder if we can get away with only replacing the first char with the results of `to_uppercase`? it seems like the second part of this is a noop?

Reply:
The comment above specifically says that everything except the first char is lowercased. (they might not be lowercase at first)

Reply:
Yeah, I thought it might be good to handle cases like: thIS iS wEiRd → This Is Weird